pkg/cluster/check/kubernetes.go

```diff
@@ -411,6 +412,11 @@ func K8sControlPlaneStaticPods(ctx context.Context, cl ClusterInfo) error {
 			resource.NewMetadata(k8s.NamespaceName, k8s.StaticPodStatusType, "", resource.VersionUndefined),
 		)
 		if err != nil {
+			if client.StatusCode(err) == codes.Unimplemented {
```

Comment:
skips the new check introduced recently for Talos < 1.2 (in the upgrade tests we run health check for v1.1)